test/v1beta1/service.go

```diff
@@ -255,3 +256,27 @@ func IsServiceReady(s *v1beta1.Service) (bool, error) {
 func IsServiceFailed(s *v1beta1.Service) (bool, error) {
 	return s.IsFailed(), nil
 }
+
+// IsServiceAndChildrenFailed will check the readiness, route and config conditions of the service
+// and return true if they are all failed.
+func IsServiceAndChildrenFailed(s *v1beta1.Service) (bool, error) {
+	if s.Generation != s.Status.ObservedGeneration {
+		return false, nil
+	}
+
+	if failed := s.IsFailed(); !failed {
+		return false, nil
```

Comment:
isn't this including the other two?

Reply:
It is, but as per the description also is already `False` if one of them is false, but we check both specifically in the test.